core/src/main/kotlin/parsers/MarkdownParser.kt

```diff
@@ -171,20 +207,84 @@ class MarkdownParser (
+                    body = text
+                        .substring(node.startOffset, node.endOffset).transform()
+                )
+                MarkdownElementTypes.MARKDOWN_FILE          -> if(node.children.size == 1) visitNode(node.children.first()) else defaultHandler(node)
                 else                                        -> defaultHandler(node)
             }
 
         private fun List<ASTNode>.evaluateChildren(): List<DocTag> =
-            this.filter { it is CompositeASTNode || it.type == MarkdownTokenTypes.TEXT }.map { visitNode(it) }
+            this.removeUselessTokens().mergeLeafASTNodes().map { visitNode(it) }
+
+        private fun List<ASTNode>.removeUselessTokens(): List<ASTNode> =
+            this.filterIndexed { index, _ ->
+                !(this[index].type == MarkdownTokenTypes.EOL &&
+                this.isLeaf(index - 1) && this.getOrNull(index - 1)?.type !in leafNodes &&
+                this.isLeaf(index + 1) && this.getOrNull(index + 1)?.type !in leafNodes) &&
+                this[index].type != MarkdownElementTypes.LINK_DEFINITION
+            }
+
+        private val notLeafNodes = listOf(MarkdownTokenTypes.HORIZONTAL_RULE)
+        private val leafNodes = listOf(MarkdownElementTypes.STRONG, MarkdownElementTypes.EMPH)
+
+        private fun List<ASTNode>.isLeaf(index: Int): Boolean =
+            if(index in 0..this.lastIndex)
+                (this[index] is CompositeASTNode)|| this[index].type in notLeafNodes
+            else
+                false
+
+        private fun List<ASTNode>.mergeLeafASTNodes(): List<ASTNode> {
+            val children: MutableList<ASTNode> = mutableListOf()
+            var index = 0
+            while(index <= this.lastIndex) {
+                if(this.isLeaf(index)) {
+                    children += this[index]
+                }
+                else {
+                    val startOffset = this[index].startOffset
+                    while(index < this.lastIndex) {
+                        if(this.isLeaf(index + 1)) {
+                            val endOffset = this[index].endOffset
+                            if(text.substring(startOffset, endOffset).transform().isNotEmpty())
+                                children += LeafASTNode(MarkdownTokenTypes.TEXT, startOffset, endOffset)
+                            break
+                        }
+                        index++
+                    }
+                    if(index == this.lastIndex) {
+                        val endOffset = this[index].endOffset
+                        if(text.substring(startOffset, endOffset).transform().isNotEmpty())
+                            children += LeafASTNode(MarkdownTokenTypes.TEXT, startOffset, endOffset)
+                    }
+                }
+                index++
+            }
+            return children
+        }
+
+        private fun String.transform() = this
+            .replace(Regex("\n\n+"), "")
+            .replace(Regex("\n>+ "), "\n")
```

Comment:
May Windows line endings be present there? If so they should also be replaced if I understand this correctly

Reply:
Tested on CRLF, did fine. We can add in the future source file test with CRLF encoded file to automate and ensure this case